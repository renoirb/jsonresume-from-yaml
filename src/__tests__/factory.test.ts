import { Resume } from '..';

// https://github.com/jsonresume/resume-schema/blob/v1.0.0/examples/valid/complete.json
// tslint:disable:object-literal-sort-keys no-any
const fixture = {
  email: 'richard.hendriks@piedpiper.com',
  label: 'Founder & CEO',
  location: {
    city: 'San Francisco',
  },
  name: 'Richard Hendriks',
  phone: '(912) 555-4321',
  summary:
    'Richard hails from Tulsa. He has earned degrees from the University of Oklahoma and Stanford.',
  url: 'https://piedpiper.com',
};

describe('Resume', () => {
  it('Has sensible default when empty constructor', () => {
    const subject = new Resume();
    expect(subject).toMatchSnapshot();
  });
  it('We can provide a partially valid constructor object', () => {
    const subject = new Resume(fixture);
    expect(subject).toMatchSnapshot();
  });
});
