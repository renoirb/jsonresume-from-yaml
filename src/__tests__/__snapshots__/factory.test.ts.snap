// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Resume Has sensible default when empty constructor 1`] = `
Resume {
  "awards": Array [],
  "basics": Object {
    "label": "Programmer",
    "name": "John Doe",
    "summary": "A summary of John Doe...",
  },
  "education": Array [],
  "interests": Array [],
  "languages": Array [],
  "publications": Array [],
  "references": Array [],
  "skills": Array [],
  "volunteer": Array [],
  "work": Array [],
}
`;

exports[`Resume We can provide a partially valid constructor object 1`] = `
Resume {
  "awards": Array [],
  "basics": Object {
    "email": "richard.hendriks@piedpiper.com",
    "label": "Founder & CEO",
    "location": Object {
      "city": "San Francisco",
    },
    "name": "Richard Hendriks",
    "phone": "(912) 555-4321",
    "summary": "Richard hails from Tulsa. He has earned degrees from the University of Oklahoma and Stanford.",
    "url": "https://piedpiper.com",
  },
  "education": Array [],
  "interests": Array [],
  "languages": Array [],
  "publications": Array [],
  "references": Array [],
  "skills": Array [],
  "volunteer": Array [],
  "work": Array [],
}
`;
